pages/survey.py

```diff
@@ -127,21 +136,29 @@ def finished():
 def video_submission():
     st.title("Video Upload")
 
+    # Initialize session state for tracking submission and upload progress
+    if "uploading" not in st.session_state:
+        st.session_state.uploading = False
+
+    if "video_submitted" not in st.session_state:
+        st.session_state.video_submitted = False  # Tracks if the video was already submitted
+
     # File uploader that only accepts video files
-    uploaded_video = st.file_uploader("Upload a video file", type=["webm"]) # "mp4", "mov", "avi", 
+    uploaded_video = st.file_uploader("Upload a video file", type=["webm"])  # You can add other file types if needed
 
     # Ensure the user uploads a video before enabling the submit button
     if uploaded_video is None:
         st.warning("Please upload a video file before proceeding.")
         st.button("Submit", disabled=True)
     else:
         try:
-            st.success("Video uploaded successfully!")            
+            st.success("Video uploaded successfully!")
             # Display the video in the app
             st.video(uploaded_video)
-            
-            if st.button("Submit", key="submit_recording", disabled=st.session_state.uploading):
-                 # Disable the button and show progress
+
+            # Disable the submit button if the video has already been submitted
+            if st.button("Submit", key="submit_recording", disabled=st.session_state.uploading or st.session_state.video_submitted):
```

Comment:
nit: consolidate the two buttons here on 160 and 152 into one bc they're basically the same button?

Reply:
I don't agree as moving the button on line 160 outside the if/else results in the button's logic not working as intended. As for line 152, it is intended more of a stylistic thing that to submit, you must submit a video.

Reply:
ah ok we can leave it as it is then. merging